src/profile-logic/tracks.js

```diff
@@ -115,6 +115,7 @@ function _getDefaultLocalTrackOrder(tracks: LocalTrack[], profile: ?Profile) {
 
     if (
       profile &&
+      !profile.meta.keepProfileThreadOrder &&
```

Comment:
I think you want to do like below, return the difference of counterIndexes. Otherwise I believe that the sort isn't stable.

Reply:
I changed the patch as you suggested to make the behavior more explicit, but I think the sort was already stable. If we didn't enter the power track or thread branch, we returned
```
return (
      LOCAL_TRACK_DISPLAY_ORDER[tracks[a].type] -
      LOCAL_TRACK_DISPLAY_ORDER[tracks[b].type]
    );
```
which would always be 0 for 2 tracks of the same type.

[MDN says](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/sort) "A comparator that always returns 0 will cause the array to not be changed at all, but is reliable nonetheless."